include/zephyr/pm/policy.h

```diff
@@ -246,6 +251,48 @@ void pm_policy_device_power_lock_get(const struct device *dev);
  */
 void pm_policy_device_power_lock_put(const struct device *dev);
 
+#ifdef CONFIG_PM_LOW_LATENCY_EVENTS
+/**
+ * @brief Register a low latency event.
+ *
+ * This is same as pm_policy_event_register but adds a low latency property
+ * to the event. The system will make sure to be up when the event occurs.
+ * For more precision, it takes as argument the absolute time in cycles
+ *
+ * @note It is mandatory to unregister events once they have happened by using
+ * pm_policy_event_unregister(). Not doing so is an API contract violation,
+ * because the system would continue to consider them as valid events in the
+ * *far* future, that is, after the cycle counter rollover.
+ *
+ * @param evt Event.
+ * @param cycle When the event will occur, in absolute time.
```

Comment:
@ankuns , @ppryga-nordic , @martintv , please have a look if this parameter is sufficient to express radio protocol timings.

Reply:
Since this is absolute time value expressed in clock cycles that can be 1MHz in case of 54H/L. It limits us to setup events not later than 71 minutes (2^32/1M = ~4294s) from SOC boot. That isn't something that we want I think.

It doesn't mean that we schedule events for e.g. 70 o more minutes ahead, but we can schedule an event 3 minutes in future when a SOC is already 69 minutes after boot.

Wouldn't it be more straight forward to use `uint64_t` for cycles count and do the same for internal implementation. Then there is less problems with handling overflows in `update_next_event`

Reply:
@ppryga-nordic if you want to schedule an event that overflows the current cycle counter it should work as well.
You just need to setup the (current_cycle + cycles_until_next_event) & MASK_32
I still don't see the limitation here.
update_next_event takes into account these overflows.
If we change the cycle count to uint64_t we still need to handle the overflows which is even more complicated.

Reply:
@rghaddab yes that is correct if the scheduled event is in future but not later than 2 times `UNIT32_MAX`. However I haven't got use case for that now, but in future that can change. 

I'm sure the implementation based on the `uint32_t` is less complex because the whole PM implementation is based on assumption the system clock counts time within 32bit span. 

Reaching a bit over the horizon.in case there is more progress in timer resolution and we have `ns` resolution timer we have to re-implement it because the clock will expire in ~4s, so it will be very easy to overlap it multiple times.

Reply:
I don't think this PR should refactor the timing subsystem in Zephyr to prepare it for nano-second timers. For micro-second timers `uint32_t` range seems good enough. I don't think anyone could need to express an event in 70 min. And even if someone did, I think waking up the system 70 min before the event just to register the event would be an acceptable workaround.

Reply:
@hubertmis this looks sufficient for 15.4.